### PR TITLE
update eslint and fix violations

### PR DESCRIPTION
**Pull request by nabdelgadir**
#### Related issues

- connect to https://github.com/strongloop-internal/scrum-apex/issues/425

### Checklist

- [ ] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

**Comment by nabdelgadir** [comment]
CI is failing but it says `No identified problem
No problems were identified. If you know why this problem occurred, please add a suitable Cause for it.` I don't believe it's related.

File: package.json [hunk 1]
```diff
@@ -48,8 +48,8 @@
     "chai": "^3.5.0",
     "chai-connect-middleware": "^0.3.1",
     "chai-oauth2orize-grant": "^0.2.0",
-    "eslint": "^2.13.1",
-    "eslint-config-loopback": "^4.0.0",
+    "eslint": "^4.18.2",
+    "eslint-config-loopback": "^8.0.0",
```
Comment (bajtos) [inline package.json]: Please update to the latest config version, which is `13.1.0` ATM.

**Comment by bajtos** [comment]
> The only change in question is whether we want to introduce `package-lock.json`.  
@raymondfeng @bajtos, what's your thought on that? Thanks.

We are intentionally NOT USING package-lock files in LB3 repositories. I am open to discuss a switch to package locking, but not in this PR please.

Please remove `package-lock.json` file from the patch.

**Comment by nabdelgadir** [comment]
@bajtos @dhmlau I didn't mean to add the `package-lock.json`s (to this one or other related PRs). They were being generated when I did `npm i` and I didn't notice them but I'll take them out of the PRs they're in. Thanks!

**Pull request merged**